presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/cache/CachingHiveMetastore.java

```diff
@@ -760,7 +728,29 @@ public void revokeRoles(Set<String> roles, Set<HivePrincipal> grantees, boolean
         return delegate.listRoleGrants(principal);
     }
 
-    private void invalidatePartitionCache(String databaseName, String tableName)
+    private void invalidateDatabase(String databaseName)
+    {
+        databaseCache.invalidate(databaseName);
+        databaseNamesCache.invalidateAll();
+    }
+
+    protected void invalidateTableWithPartitions(String databaseName, String tableName)
```

Comment:
I don't think this name change is helpful. The method `invalidateTable` is invalidating all data related to a table. The *with partitions* seems strange, since we don't have `TableWithPartitions` object. It's really doing *table and partitions*, but at that point you need to call it *table and partitions and statistics and privileges* (and views, too).